src/main/java/seedu/address/logic/parser/FindExpressionParser.java

```diff
@@ -0,0 +1,288 @@
+     * both "n/Bob" and "(t/friends AND t/colleagues)" are terms combined to form an expression.
+     * </p>
+     *
+     * @return The parsed expression node.
+     */
+    private ExprNode expression() throws ParseException {
+        ExprNode node = term();
+
+        while (match(Token.Type.OR)) {
+            Token op = consume(Token.Type.OR);
+            ExprNode right = term();
+            node = new BinaryOpNode(node, FindExpressionOperator.OR, right);
+        }
+
+        return node;
+    }
+
+    /**
+     * Parses and constructs a "term" node from the current list of tokens.
+     * <p>
+     * In the context of parsers and grammars, a "term" is a mid-level construct that consists of one or more
+     * "factors". The factors within a term are usually combined using the AND operator. For instance,
+     * in a query "n/Bob AND t/friends", both "n/Bob" and "t/friends" are factors combined to create a term.
+     * </p>
+     *
+     * @return The parsed term node.
+     */
+    private ExprNode term() throws ParseException {
+        ExprNode node = factor();
+
+        while (match(Token.Type.AND)) {
+            Token op = consume(Token.Type.AND);
+            ExprNode right = factor();
+            node = new BinaryOpNode(node, FindExpressionOperator.AND, right);
+        }
+
+        return node;
+    }
+
+    /**
+     * Parses and constructs a "factor" node from the current list of tokens.
+     * <p>
+     * A "factor" is a foundational construct in this parser. It's the most granular unit and represents
+     * a single condition or criteria in the query. Factors are building blocks for terms and expressions.
+     * </p>
+     *
+     * @return The parsed factor node.
+     */
+    private ExprNode factor() throws ParseException {
+        if (match(Token.Type.NOT)) {
+            consume(Token.Type.NOT);
+            ExprNode node = factor();
+            return new NotNode(node);
+        } else if (match(Token.Type.LPAREN)) {
+            consume(Token.Type.LPAREN);
+            ExprNode node = expression();
+            consume(Token.Type.RPAREN);
+            return node;
+        } else {
+            Token token = consume(Token.Type.CONDITION);
+            // check if text contains a slash and is a valid condition
+            if (!token.text.contains("/") || token.text.startsWith("/") || token.text.endsWith("/")) {
+                throw new ParseException("Invalid condition: " + token.text);
+            }
+            // split by slash but include slash in substrings
+            String[] parts = token.text.split("(?<=/)");
+            FindSupportedField field = FindSupportedField.fromPrefix(parts[0].trim().toLowerCase());
+            return new ConditionNode(field, parts[1].trim());
+        }
+    }
+
+    private boolean match(Token.Type type) {
+        return check(type);
+    }
+
+    private boolean check(Token.Type type) {
```

Comment:
ditto

Reply:
I agree, thanks for pointing this out! Fixed in d1ca29499770bfc0cbed8a5182530082c46154cc